packages/core/src/decorators/oas2/remove-unused-components.ts

```diff
@@ -16,15 +16,44 @@ export const RemoveUnusedComponents: Oas2Decorator = () => {
     name: string
   ): void {
     components.set(location.absolutePointer, {
-      used: components.get(location.absolutePointer)?.used || false,
+      usedIn: components.get(location.absolutePointer)?.usedIn ?? [],
       componentType,
       name,
     });
   }
 
+  function removeUnusedComponents(root: Oas2Definition, removedPaths: string[]): number {
+    const removedLengthStart = removedPaths.length;
+
+    for (const [path, { usedIn, name, componentType }] of components) {
+      const used = usedIn?.some(
+        (location) =>
+          !removedPaths.some(
+            (removed) =>
+              location.absolutePointer.startsWith(removed) &&
+              (location.absolutePointer.length === removed.length ||
```

Comment:
Isn't this the same as the following?
```
location.absolutePointer === removed ||
location.absolutePointer.startsWith(removed + '/') 

```

Reply:
Yes, but it isn't comparing the strings twice & doesn't create a string copy. We are running through this loop a lot, so making it performant seems important.

Reply:
Added a comment!